bolt-simple-taproot.md

```diff
@@ -0,0 +1,1141 @@
+```
+<remote_sig> <to_remote_script> <to_remote_control_block>
+```
+
+where `to_remote_control_block` is:
+```
+(output_key_y_parity | 0xc0) || combined_funding_key
+```
+
+#### Anchor Outputs
+
+For simple taproot channels (`option_simple_taproot`) we'll maintain the same
+form as base segwit channels, but instead will utilize `local_delayedpubkey`
+and the `remotepubkey` rather than the multi-sig keys as that's no longer
+revealed due to musig2.
+
+An anchor output has the following form:
+
+  * `OP_1 anchor_output_key`
+  * where:
+    * `anchor_internal_key = remotepubkey/local_delayedpubkey`
+    * `anchor_output_key = anchor_internal_key + tagged_hash("TapTweak", anchor_internal_key || anchor_script_root)`
+    * `anchor_script_root = tapscript_root([anchor_script])`
+    * `anchor_script`:
+        ```
+        OP_16 OP_CHECKSEQUENCEVERIFY
+        ```
+
+This output can be swept by anyone after 16 blocks with the following witness:
+```
+<anchor_script> <anchor_control_block>
+```
+
+where `anchor_control_block` is:
+```
+(output_key_y_parity | 0xc0) || anchor_internal_key
+```
+
+`nSequence` needs to be set to 16.
+
+### HTLC Scripts & Transactions
+
+#### Offered HTLCs
+
+We retain the same second-level HTLC mappings as base channels. The main
+modifications are using the revocation public key as the internal key, and
+splitting the timeout and success paths into individual script leaves.
+
+An offered HTLC has the following form:
+
+  * `OP_1 offered_htlc_key`
+  * where:
+    * `offered_htlc_key = revocation_pubkey + tagged_hash("TapTweak", revocation_pubkey || htlc_script_root)`
+    * `htlc_script_root = tapscript_root([htlc_timeout, htlc_success])`
+    * `htlc_timeout`:
+        ```
+        <local_htlcpubkey> OP_CHECKSIGVERIFY
+        <remote_htlcpubkey> OP_CHECKSIG
+        ```
+    * `htlc_success`:
+        ```
+        OP_SIZE 32 OP_EQUALVERIFY OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+        <remote_htlcpubkey> OP_CHECKSIG
+        OP_CHECKSEQUENCEVERIFY
+        ```
+
+In order to spend a offered HTLC, via either script path, an `inclusion_proof`
+must be specified along with the control block. This `inclusion_proof` is
+simply the `tap_leaf` hash of the path _not_ taken.
+
+#### Accepted HTLCs
+
+Accepted HTLCs inherit a similar format:
+
+  * `OP_1 accepted_htlc_key`
+  * where:
+    * `accepted_htlc_key = revocation_pubkey + tagged_hash("TapTweak", revocation_pubkey || htlc_script_root)`
+    * `htlc_script_root = tapscript_root([htlc_timeout, htlc_success])`
+    * `htlc_timeout`:
+        ```
+        <remote_htlcpubkey> OP_CHECKSIG
+        OP_CHECKSEQUENCEVERIFY
+        <cltv_expiry> OP_CHECKLOCKTIMEVERIFY OP_DROP
+        ```
+    * `htlc_success`:
+        ```
+        OP_SIZE 32 OP_EQUALVERIFY OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+        <local_htlcpubkey> OP_CHECKSIGVERIFY
+        <remote_htlcpubkey> OP_CHECKSIG
```

Comment:
Missing a  OP_CHECKSEQUENCEVERIFY, same as in the Offered HTLC case (timeout path)

Reply:
Not needed because the remote signature prevents you from changing it

Reply:
hmm not sure but we are talking anchor ANYONE_CANPAY signatures ? For Static Remote you are right but for Anchor-Channels its needed.

Reply:
see also [option anchors](https://github.com/lightning/bolts/blob/master/03-transactions.md#offered-htlc-outputs)

Reply:
As @antonilol said, its not needed because the signature guards the sequence of the followup transaction.